holopy/inference/result.py

```diff
@@ -187,13 +188,13 @@ def _save(self, filename, **kwargs):
     # deprecated methods as of 3.3
     def best_fit(self):
         # this method is published in the HoloPy paper
-        from holopy.fitting import fit_warning
-        fit_warning('FitResult.hologram', 'SamplingResult.best_fit()')
+        raise_fitting_api_error(
```

Comment:
I feel like any previously published code should be held onto for as long as possible in terms of either making it work, or providing an informative message (like you've done here), so I think it's good to keep this around for now.